arcade/arcade/core/executor.py

```diff
@@ -49,8 +49,17 @@ async def run(
             # serialize the output model
             output = await ToolExecutor._serialize_output(output_model, results)
 
+            # gather any warnings
+            warnings = []
+            if definition.deprecation_message:
+                warnings.append(
+                    ToolCallWarning(
+                        message=definition.deprecation_message, warning_type="deprecation"
+                    )
+                )
+
             # return the output
-            return output_factory.success(data=output)
+            return output_factory.success(data=output, warnings=warnings)
```

Comment:
only pass on the warnings if tool execution was successful. Thoughts?

Reply:
This makes sense strictly for this deprecation use case.

But thinking ahead: we will eventually want capture any number of logs (debug, info, warning, up to error) that could help someone understand why a particular tool failed. For example, a complex tool could emit logs as it steps through chunks of business logic, which would help someone pinpoint why it failed.
I'm OK if this isn't addressed in this PR, as long as it's filed for the future. 👍

Reply:
addressed